lnwire/dns_addr.go

```diff
@@ -86,3 +90,77 @@ func ValidateDNSAddr(hostname string, port uint16) error {
 
 	return nil
 }
+
+// Record returns a TLV record that can be used to encode/decode the DNSAddress.
+//
+// NOTE: this is part of the tlv.RecordProducer interface.
+func (d *DNSAddress) Record() tlv.Record {
+	sizeFunc := func() uint64 {
+		// Hostname length + 2 bytes for port.
+		return uint64(len(d.Hostname) + 2)
+	}
+
+	return tlv.MakeDynamicRecord(
+		0, d, sizeFunc, dnsAddressEncoder, dnsAddressDecoder,
+	)
+}
+
+// dnsAddressEncoder is a TLV encoder for DNSAddress.
+func dnsAddressEncoder(w io.Writer, val interface{}, _ *[8]byte) error {
+	if v, ok := val.(*DNSAddress); ok {
+		// Write the hostname as raw bytes (no length prefix for TLV).
+		if _, err := w.Write([]byte(v.Hostname)); err != nil {
+			return err
+		}
+
+		// Write the port as 2 bytes.
+		var portBytes [2]byte
+		err := WriteUint16(bytes.NewBuffer(portBytes[:0]), v.Port)
```

Comment:
hmm why `portBytes[:0]` but not `portBytes[:]`

Reply:
Totally agree that it's hard to read - so have updated with simplified version 👍 

But out of interest:

portBytes[:0] creates a slice with zero length but the same underlying capacity as portBytes. This is used to reset the buffer position for bytes.NewBuffer() to start writing at the beginning of
  the array.

  If you used portBytes[:], it would create a slice with the full length (2 bytes) containing whatever data was previously in the array, which would interfere with the WriteUint16 function.

  The [:0] slice gives bytes.NewBuffer an empty slice to start with, ensuring clean writes to the array.